contrib/package_plugin.py

```diff
@@ -0,0 +1,313 @@
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+This script automates all the work involved in packaging an external Electron Cash plugin.
+"""
+
+import compileall
+import hashlib
+import json
+import os
+import shutil
+import sys        
+import tempfile
+import traceback
+
+from PyQt5.QtGui import *
+from PyQt5.QtCore import *
+from PyQt5.QtWidgets import *
+
+## Start copied from .util
+def versiontuple(v):
+    return tuple(map(int, (v.split("."))))
+## End copied from .util
+
+def write_plugin_archive(metadata, source_package_path, archive_file_path):
+    suffixless_path = archive_file_path.strip()
+    # Ensure we don't end up with .zip.zip, as `make_archive` adds the suffix as well.
+    if suffixless_path[-4:].lower() == ".zip":
+        suffixless_path = suffixless_path[:-4]
+        
+    package_directory_name = os.path.basename(source_package_path)
+    with tempfile.TemporaryDirectory() as temp_path:
+        # Export the generated 'manifest.json' file into place.
+        manifest_file_path = os.path.join(temp_path, "manifest.json")
+        write_manifest(metadata, manifest_file_path)
+
+        # Copy the selected Python package into place.
+        dest_package_path = os.path.join(temp_path, package_directory_name)
+        shutil.copytree(source_package_path,  dest_package_path)
+        # Python bytecode cannot be written into the zip archive as Electron Cash runs it.
+        # So we precompile it before creating the archived form.
+        compileall.compile_dir(dest_package_path)           
+        shutil.make_archive(suffixless_path, 'zip', temp_path)
+        
+    plugin_path = suffixless_path +".zip"
+    hash_md5 = hashlib.md5()
+    with open(plugin_path, "rb") as f:
+        hash_md5.update(f.read())
+    return hash_md5.hexdigest()
+
+def write_manifest(metadata, manifest_file_path):
+    with open(manifest_file_path, "w") as f:
+        json.dump(metadata, f, indent=4)
+
+def build_manifest(display_name, version, description, minimum_ec_version, package_name, available_for_qt, available_for_cmdline):
+    metadata = {}
+    metadata["display_name"] = display_name
+    version_value = version.strip()
+    try:
+        versiontuple(version_value)
+    except ValueError:
+        version_value = "0.0"
+    metadata["version"] = version_value
+    metadata["description"] = description
+    version_value = minimum_ec_version.strip()
+    try:
+        versiontuple(version_value)
+    except ValueError:
+        version_value = "0.0"
+    metadata["minimum_ec_version"] = version_value
+    if package_name is not None:
+        metadata["package_name"] = package_name
+    available_for = []
+    if available_for_qt:
+        available_for.append("qt")
+    if available_for_cmdline:
+        available_for.append("cmdline")
+    metadata["available_for"] = available_for
```

Comment:
kivy.  I left it out on purpose.   Later versions of this will both handle further more types, and other more advanced functionality (like bundling images in the zip archive).  For now it serves as an illustration of how to do it in the main cases.